### PR TITLE
doc: meeting notes from 2016-12-01

### PR DESCRIPTION
**Pull request by joshgav**
##### Checklist

- [x] documentation is changed or added
- [x] the commit message follows commit guidelines

##### Affected core subsystem(s)

doc

##### Description of change

notes from Collaboration Summit meeting

/cc @nodejs/api @nodejs/chakracore @nodejs/v8 @nodejs/spidernode @nodejs/ctc  @nodejs/tsc

### CODE REVIEW

**Review by jasnell — APPROVED** [review]

File: meeting-notes/2016-12-01.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+@jasnell: Will V8 have something usable by v8.x (April)?
+
+@s3ththompson: Think so, need to check with Ali (@ofrobots).
+
+@mhdawson: Would be good to have Chakra team review FFI work in V8.
+
+**Next steps**:
+
+* V8 to publish branch with Fast FFI implementation.
```
Comment (aruneshchandra) [inline meeting-notes/2016-12-01.md]: Also discussed was for ChakraCore / SpiderMonkey to provide feedback on the FFI implementation.  Maybe collapse both the next steps into one ?

Comment (aruneshchandra) [reply meeting-notes/2016-12-01.md]: And also Google to provide feedback on NAPI

**Review by aruneshchandra — APPROVED** [review]
Other than capturing additional next steps., this is great!

**Review by jasnell — APPROVED** [review]

**Comment by joshgav** [comment]
@aruneshchandra I cleaned up inline next steps and provided a consolidated list at the end, PTAL. Copied here for better visibility:

* CTC to review [Node-EP for abi-stable-node](https://github.com/nodejs/node-eps/pull/20).
* V8 team to review [NAPI implementation](https://github.com/nodejs/abi-stable-node).
* NAPI team to test against more native modules in CITGM.
* Core/CTC to land as experimental behind a build-time flag in 8.0; and to evaluate bringing out from flag when 8.0 goes LTS (Oct 2017).
* V8 to publish branch with Fast FFI implementation; Node Core and Chakra teams to review.
* API WG to hold checkpoint with CTC in late February/early March 2017, demo and discuss NAPI and Fast-FFI.
* Checkpoint again in mid-July, perhaps in conjunction with a community conference.

@mhdawson @ofrobots could I get a review from you too? @ofrobots you weren't present at the meeting, but can you confirm the V8 team is on board with these next steps? [Note: @s3ththompson was present.] Thank you!

File: meeting-notes/2016-12-01.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+@mhdawson: Depends on needs of Node and each VM.
+
+We provide an `Environment*` pointer in each NAPI API, which gives deeper access
+if necessary.
+
+@aruneshchandra: How do we ensure buy-in from VM owners?
+
+@rvagg: If NAPI is accepted downstream VM owners will have to participate.
+
+@s3ththompson: V8 team has reviewed NAPI work and doesn’t have major concerns,
```
Comment (s3ththompson) [inline meeting-notes/2016-12-01.md]: Thanks for writing up these notes! This summary doesn't quite capture what I was trying to convey.  Perhaps: "the V8 team haven't yet uncovered any major issues, but we also haven't done a full review of the proposal."

I think it's under next steps below that the V8 team needs to do a review of the API.

cc @jeisinger

File: meeting-notes/2016-12-01.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+@rvagg: Would be good to rephrase from “VM Neutrality,” which is somewhat
+controversial, to “ABI Stable Node”, which is what is most important to users.
+
+@mhdawson: Tested some sizeable modules, perf is within reasonable limits. Next
+step is error handling. Also need more feedback from community.
+
+API is C, some VMs use C++. Goal is to ensure C API and ABI are stable. C++
+wrappers/helpers may not be as stable. We’d like to add some common C++-style
+stuff, like `NAN::New`.
```
Comment (mhdawson) [inline meeting-notes/2016-12-01.md]: API is C as we don't believe a C++ API can be as ABI stable.  However, we do want to add some C++ sugar that would not be part of the ABI stable API but which would be inlined and only call the C API.  A good example would be `NAN::New`.

File: meeting-notes/2016-12-01.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+work specific to V8.
+
+@jasnell: Are there things in Node which should also be part of this API?
+
+@mhdawson: Some libuv abstractions were already in NAN, learning from some of
+those.
+
+Buffer is another example.
+
+@aruneshchandra: Thinking of trying with native modules in CITGM.
```
Comment (mhdawson) [inline meeting-notes/2016-12-01.md]: @aruneshchandra: our plan is to review the native modules in CITGM to validate they don't use anything outside of the existing API.

File: meeting-notes/2016-12-01.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+review of Node-EP.
+
+@jasnell: What do we need to achieve to consider NAPI feature-complete? All of
+V8 API?
+
+@aruneshchandra: No, not all of V8, but all that is used by native modules.
+
+@rvagg: But modules may use a lot of those APIs (e.g. garbage collection).
+
+@mhdawson: We’re assuming that’s a low number and we won’t address those.
```
Comment (mhdawson) [inline meeting-notes/2016-12-01.md]: +@mhdawson: We’re assuming that’s a low number and we won’t address those in the first version.

**Review by mhdawson — APPROVED** [review]
LGTM with the suggested changes.

**Comment by fhinkel** [comment]
Sorry to be behind on this. We're reviewing the NAPI right now. Is it OK if we have results next week? /cc @nodejs/v8

**Comment by aruneshchandra** [comment]
@fhinkel yes its ok! BTW you are not behind :)

**Comment by joshgav** [comment]
Updated, but I don't have push access in this repo. @mhdawson can you land? Thanks!

**Comment by aruneshchandra** [comment]
Landed it! Thanx @joshgav

**Comment by fhinkel** [comment]
Thanks for getting the NAPI in such good shape! No major concerns from our side, I've opened a few minor PRs (https://github.com/nodejs/abi-stable-node/pull/32, https://github.com/nodejs/abi-stable-node/pull/33, https://github.com/nodejs/abi-stable-node/pull/34). 

Along the lines of to the comment in the [header](https://github.com/nodejs/abi-stable-node/blob/api-prototype-6.2.0/src/node_jsvmapi.h#L10), experience with our API and Blink suggests all hat wrapping will come at a cost. Eager to see a performance analysis. 

/cc @ofrobots

**Pull request merged**